src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -703,11 +709,13 @@ public static IEnumerable<TSource> SkipWhile<TSource>(this IEnumerable<TSource>
 
         private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            bool yielding = false;
-            foreach (TSource element in source)
+            using (IEnumerator<TSource> e = source.GetEnumerator())
```

Comment:
@leppie, please refrain from profanity.  And that's not what he's saying: he's saying that there's a compiler error here resulting from an undefined variable being used, and he was under the impression that ```element``` used later and ```e``` used here were meant to be the same variable (that's not actually the case, though).